src/platform/packages/shared/kbn-coloring/src/shared_components/color_mapping/color/rule_matching.ts

```diff
@@ -7,33 +7,53 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
+import { RawValue, deserializeField } from '@kbn/data-plugin/common';
 import { i18n } from '@kbn/i18n';
 import { ColorMapping } from '../config';
 
-export function ruleMatch(
-  rule: ColorMapping.Config['assignments'][number]['rule'],
-  value: string | number | string[]
-) {
-  switch (rule.type) {
-    case 'matchExactly':
-      if (Array.isArray(value)) {
-        return rule.values.some(
-          (v) =>
-            Array.isArray(v) && v.length === value.length && v.every((part, i) => part === value[i])
-        );
+/**
+ * Predicate used to match an `Assignment` to a given raw value
+ */
+export function assignmentMatchFn(rawValue: RawValue) {
+  const ruleMatch = ruleMatchFn(rawValue);
+  return function assignmentMatch(assignment: ColorMapping.Assignment) {
+    return assignment.rules.some((rule) => ruleMatch(rule));
+  };
+}
+
+/**
+ * Predicate used to match an `Assignment` to a given raw value
+ */
+export function ruleMatchFn(rawValue: RawValue) {
+  return function ruleMatch(rule: ColorMapping.ColorRule) {
```

Comment:
Probably you don't need currying here because the way you areusing this you can just pass both arguments in the ruleMatch function. Also no need to export this function as it doesn't seems to be used elsewhere

Reply:
Sorry, this file was replaced with the simplified `ColorAssignmentMatcher` and [`match.tsx`](https://github.com/elastic/kibana/pull/207957/files#diff-91b91f6ad40b0a5ef4537c20180c7683b2d3620f3a70b14ffc0a0d458c208e5e). Removed in f1ca760436d1f7ba2b1c6bdfd67e6e8d9599385e.

Comment no longer applies.